src/components/organisms/Header.jsx

```diff
@@ -0,0 +1,110 @@
+import React, { useEffect, useState } from 'react';
+import { NavLink } from 'react-router-dom';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faBook, faBars } from '@fortawesome/free-solid-svg-icons';
+import UserAuthBtn from '../molecules/UserAuthBtn';
+
+function Header() {
+  const [collapse, setCollapse] = useState();
+  const [toggleClass, setToggleClass] = useState('');
+  const [toggleAria, setToggleAria] = useState('');
+  const [navClass, setNavClass] = useState('');
+
+  const updateToggleClass = (isActive) => {
+    if (isActive) {
+      setToggleClass('navbar-toggler collapsed');
```

Comment:
You should hard code `navbar-toggler` into your classes. So the JSX looks like this. No reason to set the same thing when it can be static.

```
className{`${toggleClass} navbar-toggler p-1`}
```